File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/TenerService.java
package com.taekwondo.service;
import java.util.List;

import com.taekwondo.model.Tener;
import com.taekwondo.model.Tipo_Evento;
import com.taekwondo.model.Evento;

public interface TenerService {
	public abstract List<Tener> getTener();
	public abstract List<Evento> getEventosConTipo(int id_tipo);
	public abstract List<Tipo_Evento> getTiposDelEvento(int id_evento);
	public abstract Tener getTener(int id_tipo, int id_evento);
	public abstract void createTener(Tener t);
	public abstract void deleteTener(int id_tener);
	

}

File: proyectoTaeKwonDo/src/app/_services/tener.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Tener } from '../_models/tener';

@Injectable({
  providedIn: 'root'
})
export class TenerService {

  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getTener(){
    return this.http.get(this.API_URI+'/tener');
  }

  getEventosDeTipo(id_tipo_evento: number){
    return this.http.get(this.API_URI+'/tener/tipo_evento/'+id_tipo_evento);
  }

  getTiposDelEvento(id_evento: number){
    return this.http.get(this.API_URI+'/tener/evento/'+id_evento);
  }

  createTener(tener: Tener){
    return this.http.post(this.API_URI+'/tener', tener);
  }

  deleteTener(id_evento: number, id_tipo: number){
    return this.http.delete(this.API_URI+'/tener/'+id_evento+'/'+id_tipo);
  }
}
